### PR TITLE
Remove the Mutex in Buffer

### PR DESCRIPTION
**Pull request by ekexium**
I don't see the reason of having a Mutex here... Do I miss anything?

### CODE REVIEW

File: src/transaction/buffer.rs [hunk 1]
```diff
@@ -34,24 +33,24 @@ impl InnerBuffer {
...
 pub struct Buffer {
-    mutations: Mutex<InnerBuffer>,
+    inner: InnerBuffer,
 }
 
 impl Buffer {
     pub fn new(is_pessimistic: bool) -> Buffer {
         Buffer {
-            mutations: Mutex::new(InnerBuffer::new(is_pessimistic)),
+            inner: InnerBuffer::new(is_pessimistic),
```
Comment (nrc) [inline src/transaction/buffer.rs:42]: If we don't have the Mutex, then I don't think we need the inner/outer split.

**Review by nrc — COMMENTED** [review]
Nice, that this works, good idea to try it!

File: src/transaction/buffer.rs (hunk 1 above)
Comment (ekexium) [reply src/transaction/buffer.rs:42]: Yes. I hoped to make the PR simple :). Will update it.

**Review by nrc — APPROVED** [review]
lgtm!

**Pull request merged**